packages/userstorage/src/browser/user-storage-resource.ts

```diff
@@ -15,15 +15,10 @@ export class UserStorageResource implements Resource {
 
     protected readonly onDidChangeContentsEmitter = new Emitter<void>();
     protected readonly toDispose = new DisposableCollection();
-    private readonly _uri: string;
-
     constructor(
         public uri: URI,
         protected readonly service: UserStorageService
     ) {
-
-        this._uri = uri.toString();
-
         this.toDispose.push(this.service.onUserStorageChanged(e => {
             for (const changedUri of e.uris) {
                 if (changedUri === this.uri) {
```

Comment:
should be compared as strings, not reference

Reply:
Oh yes, otherwise they'll never be "equal".

Reply:
@akosyakov I've fixed it and added tests for user-storage resource (which includes this kind of mistake) thanks!